packages/glimmer-runtime/tests/initial-render-test.ts

```diff
@@ -790,3 +791,45 @@ test("Namespaced attribute with unquoted expression throws", function() {
     compile("<svg xlink:title={{title}}>content</svg>");
   }, /Namespaced attributes cannot be set as props. Perhaps you meant xlink:title="{{title}}"/);
 });
+
+let warnings = 0;
+let orignalAttributeFor;
+
+const StyleAttribute = {
+  setAttribute(dom, element, attr, value) {
+    warnings++;
+    Attribute.setAttribute(dom, element, attr, value);
+  },
+  updateAttribute() {}
+}
+
+QUnit.module('Style attributes', {
+  setup() {
+    commonSetup();
+    orignalAttributeFor = env.attributeFor;
+    env.attributeFor = function(element, attr, reference) {
+      if (attr === 'style') {
+        return new ElementOperation(element, StyleAttribute, attr, reference as Reference<string>);
+      }
+      return orignalAttributeFor.call(env, element, attr, reference);
+    }
```

Comment:
Should subclass here. You can make `commonSetup` optionally take an environment instance (`commonSetup(env = new TestEnvironment()) { ... }`)